From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Szabolcs Hubai <szab.hu@gmail.com>
Date: Thu, 20 Oct 2022 02:31:28 +0200
Subject: [PATCH 3/3] ramips: rt305x: add HooToo Tripmate HT-TM01

as a HT-TM02 clone ;)
- it doesn't have a mode switch
- it does have a battery and an i2c but it is not really usable
---
 .../ramips/dts/rt5350_hootoo_ht-tm01.dts      |  37 ++++++
 .../ramips/dts/rt5350_hootoo_ht-tm02.dts      | 107 ++----------------
 .../ramips/dts/rt5350_sunvalley_tripmate.dtsi |  98 ++++++++++++++++
 target/linux/ramips/image/rt305x.mk           |  18 ++-
 .../rt305x/base-files/etc/board.d/01_leds     |   4 +
 .../rt305x/base-files/etc/board.d/02_network  |   1 +
 .../etc/uci-defaults/05_fix-compat-version    |   1 +
 .../rt305x/base-files/lib/upgrade/platform.sh |   2 +
 8 files changed, 165 insertions(+), 103 deletions(-)
 create mode 100644 target/linux/ramips/dts/rt5350_hootoo_ht-tm01.dts
 create mode 100644 target/linux/ramips/dts/rt5350_sunvalley_tripmate.dtsi

diff --git a/target/linux/ramips/dts/rt5350_hootoo_ht-tm01.dts b/target/linux/ramips/dts/rt5350_hootoo_ht-tm01.dts
new file mode 100644
index 0000000000..eff5ce7528
--- /dev/null
+++ b/target/linux/ramips/dts/rt5350_hootoo_ht-tm01.dts
@@ -0,0 +1,37 @@
+#include "rt5350_sunvalley_tripmate.dtsi"
+
+/ {
+	compatible = "hootoo,ht-tm01", "ralink,rt5350-soc";
+	model = "HooToo HT-TM01";
+
+	aliases {
+		led-boot = &led_wlan;
+		led-failsafe = &led_wlan;
+		led-running = &led_wlan;
+		led-upgrade = &led_wlan;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_wlan: wlan {
+			label = "green:wlan";
+			gpios = <&gpio0 7 GPIO_ACTIVE_LOW>;
+		};
+
+		lan {
+			label = "blue:lan";
+			gpios = <&gpio0 12 GPIO_ACTIVE_LOW>;
+		};
+	};
+};
+
+
+&i2c {
+	status = "okay";
+};
+
+&sunvalley_gpio {
+	ralink,group = "jtag", "uartf";
+	ralink,function = "gpio";
+};
diff --git a/target/linux/ramips/dts/rt5350_hootoo_ht-tm02.dts b/target/linux/ramips/dts/rt5350_hootoo_ht-tm02.dts
index 1ab3d20081..0efecc0fc8 100644
--- a/target/linux/ramips/dts/rt5350_hootoo_ht-tm02.dts
+++ b/target/linux/ramips/dts/rt5350_hootoo_ht-tm02.dts
@@ -1,7 +1,4 @@
-#include "rt5350.dtsi"
-
-#include <dt-bindings/gpio/gpio.h>
-#include <dt-bindings/input/input.h>
+#include "rt5350_sunvalley_tripmate.dtsi"
 
 / {
 	compatible = "hootoo,ht-tm02", "ralink,rt5350-soc";
@@ -27,103 +24,13 @@
 			gpios = <&gpio0 12 GPIO_ACTIVE_LOW>;
 		};
 	};
-
-	keys {
-		compatible = "gpio-keys-polled";
-		poll-interval = <20>;
-
-		reset {
-			label = "reset";
-			gpios = <&gpio0 10 GPIO_ACTIVE_LOW>;
-			linux,code = <KEY_RESTART>;
-		};
-
-		modeswitch {
-			label = "modeswitch";
-			gpios = <&gpio0 14 GPIO_ACTIVE_LOW>;
-			linux,code = <BTN_0>;
-			linux,input-type = <EV_SW>;
-		};
-	};
 };
 
-&spi0 {
-	status = "okay";
-
-	flash@0 {
-		compatible = "jedec,spi-nor";
-		reg = <0>;
-		spi-max-frequency = <50000000>;
-
-		partitions {
-			compatible = "fixed-partitions";
-			#address-cells = <1>;
-			#size-cells = <1>;
-
-			partition@0 {
-				label = "u-boot";
-				reg = <0x0 0x30000>;
-				read-only;
-			};
-
-			partition@30000 {
-				label = "config";
-				reg = <0x30000 0x10000>;
-				read-only;
-			};
-
-			factory: partition@40000 {
-				label = "factory";
-				reg = <0x40000 0x10000>;
-				read-only;
-			};
-
-			partition@50000 {
-				label = "kernel";
-				reg = <0x50000 0x180000>;
-			};
-
-			partition@1d0000 {
-				label = "u-boot-env";
-				reg = <0x1d0000 0x10000>;
-			};
-
-			partition@1e0000 {
-				label = "userbackup";
-				reg = <0x1e0000 0x10000>;
-				read-only;
-			};
-
-			partition@1f0000 {
-				label = "user";
-				reg = <0x1f0000 0x10000>;
-				read-only;
-			};
-
-			rootfs1: partition@200000 {
-				label = "rootfs";
-				reg = <0x200000 0x600000>;
-			};
-		};
-	};
-};
-
-&state_default {
-	gpio {
-		groups = "i2c", "jtag", "uartf";
-		function = "gpio";
+&sunvalley_keys {
+	modeswitch {
+		label = "modeswitch";
+		gpios = <&gpio0 14 GPIO_ACTIVE_LOW>;
+		linux,code = <BTN_0>;
+		linux,input-type = <EV_SW>;
 	};
 };
-
-&ethernet {
-	mtd-mac-address = <&factory 0x28>;
-};
-
-&esw {
-	mediatek,portmap = <0x10>;
-	mediatek,portdisable = <0x2f>;
-};
-
-&wmac {
-	ralink,mtd-eeprom = <&factory 0x0>;
-};
diff --git a/target/linux/ramips/dts/rt5350_sunvalley_tripmate.dtsi b/target/linux/ramips/dts/rt5350_sunvalley_tripmate.dtsi
new file mode 100644
index 0000000000..312772232e
--- /dev/null
+++ b/target/linux/ramips/dts/rt5350_sunvalley_tripmate.dtsi
@@ -0,0 +1,98 @@
+#include "rt5350.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	sunvalley_keys: keys {
+		compatible = "gpio-keys-polled";
+		poll-interval = <20>;
+
+		reset {
+			label = "reset";
+			gpios = <&gpio0 10 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+};
+
+&spi0 {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <50000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0 0x30000>;
+				read-only;
+			};
+
+			partition@30000 {
+				label = "config";
+				reg = <0x30000 0x10000>;
+				read-only;
+			};
+
+			factory: partition@40000 {
+				label = "factory";
+				reg = <0x40000 0x10000>;
+				read-only;
+			};
+
+			partition@50000 {
+				label = "kernel";
+				reg = <0x50000 0x180000>;
+			};
+
+			partition@1d0000 {
+				label = "u-boot-env";
+				reg = <0x1d0000 0x10000>;
+			};
+
+			partition@1e0000 {
+				label = "userbackup";
+				reg = <0x1e0000 0x10000>;
+				read-only;
+			};
+
+			partition@1f0000 {
+				label = "user";
+				reg = <0x1f0000 0x10000>;
+				read-only;
+			};
+
+			rootfs1: partition@200000 {
+				label = "rootfs";
+				reg = <0x200000 0x600000>;
+			};
+		};
+	};
+};
+
+&state_default {
+	sunvalley_gpio: gpio {
+		groups = "i2c", "jtag", "uartf";
+		function = "gpio";
+	};
+};
+
+&ethernet {
+	mtd-mac-address = <&factory 0x28>;
+};
+
+&esw {
+	mediatek,portmap = <0x10>;
+	mediatek,portdisable = <0x2f>;
+};
+
+&wmac {
+	ralink,mtd-eeprom = <&factory 0x0>;
+};
diff --git a/target/linux/ramips/image/rt305x.mk b/target/linux/ramips/image/rt305x.mk
index f8e962b425..f1debab6da 100644
--- a/target/linux/ramips/image/rt305x.mk
+++ b/target/linux/ramips/image/rt305x.mk
@@ -594,11 +594,9 @@ define Device/hilink_hlk-rm04
 endef
 TARGET_DEVICES += hilink_hlk-rm04
 
-define Device/hootoo_ht-tm02
+define Device/sunvalley_tripmate_common
   SOC := rt5350
   IMAGE_SIZE := 7872k
-  DEVICE_VENDOR := HooToo
-  DEVICE_MODEL := HT-TM02
   DEVICE_PACKAGES := kmod-usb-ohci kmod-usb2 kmod-usb-ledtrig-usbport
   IMAGE_SIZE := 6144k
   KERNEL_SIZE := 1536k
@@ -611,6 +609,20 @@ define Device/hootoo_ht-tm02
 	Upgrade via sysupgrade mechanism is not possible, so new installation via TFTP is required.
   DEFAULT := n
 endef
+
+define Device/hootoo_ht-tm01
+  $(Device/sunvalley_tripmate_common)
+  DEVICE_VENDOR := HooToo
+  DEVICE_MODEL := HT-TM01
+  DEVICE_PACKAGES += kmod-i2c-ralink
+endef
+TARGET_DEVICES += hootoo_ht-tm01
+
+define Device/hootoo_ht-tm02
+  $(Device/sunvalley_tripmate_common)
+  DEVICE_VENDOR := HooToo
+  DEVICE_MODEL := HT-TM02
+endef
 TARGET_DEVICES += hootoo_ht-tm02
 
 define Device/huawei_d105
diff --git a/target/linux/ramips/rt305x/base-files/etc/board.d/01_leds b/target/linux/ramips/rt305x/base-files/etc/board.d/01_leds
index d29328e4b0..8e4b722edd 100755
--- a/target/linux/ramips/rt305x/base-files/etc/board.d/01_leds
+++ b/target/linux/ramips/rt305x/base-files/etc/board.d/01_leds
@@ -72,6 +72,10 @@ planex,mzk-w300nh2)
 hauppauge,broadway)
 	ucidef_set_led_netdev "wifi_led" "wifi" "red:wps_active" "wlan0"
 	;;
+hootoo,ht-tm01)
+	ucidef_set_led_netdev "eth" "Ethernet" "blue:lan" "eth0"
+	ucidef_set_led_netdev "wifi_led" "wifi" "green:wlan" "wlan0"
+	;;
 hootoo,ht-tm02)
 	ucidef_set_led_netdev "eth" "Ethernet" "green:lan" "eth0"
 	ucidef_set_led_netdev "wifi_led" "wifi" "blue:wlan" "wlan0"
diff --git a/target/linux/ramips/rt305x/base-files/etc/board.d/02_network b/target/linux/ramips/rt305x/base-files/etc/board.d/02_network
index c3e39e76d0..e717795574 100755
--- a/target/linux/ramips/rt305x/base-files/etc/board.d/02_network
+++ b/target/linux/ramips/rt305x/base-files/etc/board.d/02_network
@@ -22,6 +22,7 @@ ramips_setup_interfaces()
 	hame,mpr-a1|\
 	hame,mpr-a2|\
 	hauppauge,broadway|\
+	hootoo,ht-tm01|\
 	hootoo,ht-tm02|\
 	huawei,d105|\
 	intenso,memory2move|\
diff --git a/target/linux/ramips/rt305x/base-files/etc/uci-defaults/05_fix-compat-version b/target/linux/ramips/rt305x/base-files/etc/uci-defaults/05_fix-compat-version
index 3995a2731c..698cb3fb48 100644
--- a/target/linux/ramips/rt305x/base-files/etc/uci-defaults/05_fix-compat-version
+++ b/target/linux/ramips/rt305x/base-files/etc/uci-defaults/05_fix-compat-version
@@ -1,6 +1,7 @@
 . /lib/functions.sh
 
 case "$(board_name)" in
+	hootoo,ht-tm01|\
 	hootoo,ht-tm02)
 		uci set system.@system[0].compat_version="2.0"
 		uci commit system
diff --git a/target/linux/ramips/rt305x/base-files/lib/upgrade/platform.sh b/target/linux/ramips/rt305x/base-files/lib/upgrade/platform.sh
index fec4b6a304..4903a093e4 100755
--- a/target/linux/ramips/rt305x/base-files/lib/upgrade/platform.sh
+++ b/target/linux/ramips/rt305x/base-files/lib/upgrade/platform.sh
@@ -7,6 +7,7 @@ REQUIRE_IMAGE_METADATA=1
 
 platform_check_image() {
 	case "$(board_name)" in
+	*ht-tm01|\
 	*ht-tm02)
 		local config_size=$(mtd_get_part_size config)
 
@@ -47,6 +48,7 @@ platform_do_upgrade() {
 	local board=$(board_name)
 
 	case "$board" in
+	hootoo,ht-tm01|\
 	hootoo,ht-tm02)
 		hootoo_do_upgrade "$1"
 		;;
